### PR TITLE
Add denon upgrade command

### PR DESCRIPTION
**Pull request by MoonSupport**
denon upgrade #26 

Add denon upgrade command!!

I'm a noob so I can make mistakes.

I'd love to give feedback at any time.

I hope it helps you

Thanks for a good opportunity :)

### CODE REVIEW

File: cli.ts [hunk 1]
```diff
@@ -38,10 +39,12 @@ export function help() {
       -w, --watch             List of paths to watch separated by commas
           --fmt               Adds a deno fmt executor
           --test              Adds a deno test executor
+      -u, --upgrade           Upgrade Specific Version. Default is latest.
```
Comment (eliassjogreen) [inline cli.ts:42]: Upgrade shouldn't be a flag

File: cli.ts [hunk 2]
```diff
@@ -68,6 +71,22 @@ function convertToStringArray(
...
 
+export function upgrade(version: string) {
+  const versionWithoutVprefix = RegExp(/^\d\.\d\.\d/);
+  const redColor = '\u001b[31m'
+  if(version === 'latest') {
+    version = 'master'
+  }
+
+  if(versionWithoutVprefix.test(version)) {
+    console.warn(redColor,`Do you mean v${version}? If it's you should add v prefix`)
```
Comment (eliassjogreen) [inline cli.ts:82]: Add a warn function to the `log.ts` file instead, also red is used for errors currently maybe orange would be a good alternative color. Also use the colors from the `std/fmt` module

File: denon_config.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export interface DenonConfig {
   execute: { [extension: string]: string[] };
   fmt: boolean;
   test: boolean;
+  upgrade: string;
```
Comment (eliassjogreen) [inline denon_config.ts:18]: Shouldn't be part of the config just like `--help` isn't

**Comment by eliassjogreen** [comment]
Thanks for your contribution!

In the file `cli.ts` in the switch statement on line 100 you need to add a case for `upgrade` that can take the next unknown sent in as a version if supplied.

**Comment by eliassjogreen** [comment]
Also remember to run `deno fmt`

**Comment by MoonSupport** [comment]
> Thanks for your contribution!
> 
> In the file `cli.ts` in the switch statement on line 100 you need to add a case for `upgrade` that can take the next unknown sent in as a version if supplied.

I accepted feedbacks :)
but I couldn't try above request. :(

I have two ideas, Could you check it?

1. Change Switch args to Regex from string. So check version.
2. Using github api to get tag_name.

And If not exist, upgrade latest.

Please let me know if there is a better way 🙏

File: cli.ts (hunk 1 above)
Comment (MoonSupport) [reply cli.ts]: remove flag!

File: cli.ts (hunk 2 above)
Comment (MoonSupport) [reply cli.ts]: Good

File: denon_config.ts (hunk 3 above)
Comment (MoonSupport) [reply denon_config.ts]: remove!

**Pull request closed without merging**